src/ripple/beast/core/LexicalCast.h

```diff
@@ -78,8 +78,12 @@ struct LexicalCast<Out, std::string>
     std::enable_if_t<
         std::is_integral_v<Integral> && !std::is_same_v<Integral, bool>,
         bool>
-    operator()(Integral& out, std::string const& in) const
+    operator()(Integral& out, std::string_view const& in) const
     {
+        // the underlying data of the std::string_view is no longer alive
+        if (in.empty())
+            return false;
```

Comment:
That's not what `empty()` tests. `empty()` returns whether `size() == 0`. There is no test for whether the memory pointed by `std::string_view` is still valid, and even if there were, no such test is necessary in this function. All current callers are passing `std::string`, which is guaranteed to point to valid memory. `std::string` is implicitly convertible to a `std::string_view`, which means no existing callers need to be changed. Only _future_ callers need to be aware that the `std::string_view` they are passing to this function must point to valid memory, but why wouldn't they know that? What did they think would happen to the view they pass? Please remove this condition.

Reply:
thanks for the explanation, fixed in 85e001dcb21516ad0d711e60af70fc9141f29162